### ISSUE TITLE
[naming-convention] allow all caps for TS declaration for environment variables

### ISSUE THREAD

**Issue by lensbart** [typescript-eslint/typescript-eslint#3451]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/naming-convention": ["warn"]
  }
}
```

```TS
// process.env.d.ts
// based on https://stackoverflow.com/a/53981706
declare global {
  namespace NodeJS {
    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
    interface ProcessEnv {
      NODE_ENV: 'development' | 'production'
      DATABASEURL: string
    }
  }
}

export {} // If this file has no import/export statements (i.e. is a script) convert it into a module by adding an empty export statement.
```

**Expected Result**

No warning 

**Actual Result**

Warnings for `NODE_ENV` and `DATABASEURL`

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.17.0` |
| `@typescript-eslint/parser`        | `4.17.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.21.0` |
| `node`                             | `14.15.4` |
